### ISSUE TITLE
[BUG] TypeError on Latest Version 3.14.2

### ISSUE THREAD

**Issue by coltonhurst** [francoismassart/eslint-plugin-tailwindcss#317]
**Describe the bug**

Hello! Thank you for all of the great work on this plugin.

When running `npm run lint` on version `3.14.2`, I receive the following TypeError:

```
TypeError: Cannot convert undefined or null to object
Occurred while linting /Users/coltonhurst/github/bitwarden/clients/apps/browser/src/auth/popup/account-switching/account-switcher.component.html:5
Rule: "tailwindcss/enforces-shorthand"
    at Function.keys (<anonymous>)
    at /Users/coltonhurst/github/bitwarden/clients/node_modules/eslint-plugin-tailwindcss/lib/rules/enforces-shorthand.js:248:37
    at Array.filter (<anonymous>)
    at parseForShorthandCandidates (/Users/coltonhurst/github/bitwarden/clients/node_modules/eslint-plugin-tailwindcss/lib/rules/enforces-shorthand.js:239:52)
    at attributeVisitor (/Users/coltonhurst/github/bitwarden/clients/node_modules/eslint-plugin-tailwindcss/lib/rules/enforces-shorthand.js:479:9)
    at ruleErrorHandler (/Users/coltonhurst/github/bitwarden/clients/node_modules/eslint/lib/linter/linter.js:1076:28)
    at /Users/coltonhurst/github/bitwarden/clients/node_modules/eslint/lib/linter/safe-emitter.js:45:58
    at Array.forEach (<anonymous>)
    at Object.emit (/Users/coltonhurst/github/bitwarden/clients/node_modules/eslint/lib/linter/safe-emitter.js:45:38)
    at NodeEventGenerator.applySelector (/Users/coltonhurst/github/bitwarden/clients/node_modules/eslint/lib/linter/node-event-generator.js:297:26)
```

This was discovered and is visible through this [PR](https://github.com/bitwarden/clients/pull/7936).

**To Reproduce**

1. Clone the Bitwarden clients repository [here](https://github.com/bitwarden/clients)
2. Run `npm install`
3. Run `npm ci`
4. Run `npm run lint`
  - Note: In our `package.json` we have `lint` defined as so: `"lint": "eslint . --cache --cache-strategy content && prettier --check .",`

**Expected behavior**

That `npm run lint` runs successfully.

**Screenshots**

![image](https://github.com/francoismassart/eslint-plugin-tailwindcss/assets/5691612/a82dbcae-ca6d-42fd-bed5-b2b62beedb71)

**Environment (please complete the following information):**
 - OS: macOS Sonoma `14.3.1`
 - Softwares + version used:
     -  VSCode `1.86.1`
     - Local node version is: `v18.16.0`

Thank you!

**Comment by francoismassart** [comment]
Hi @coltonhurst 
I think it was due to the fact that I was always expecting to get a value from `theme.size` which may not exists depending on your version of Tailwind CSS... Your old MR code is not available however, I did:

1. clone the latest version of the repo
2. install dependencies
3. `npm run lint` ✅ 
4. `npm i eslint-plugin-tailwindcss@3.15.0`
5. `npm run lint` ✅ 

Maybe it was fixed by the update of Tailwind CSS but I'll patch `v3.15.1` to address the issue.

I would be curious to investigate more on this. Do you still have your old PR somewhere.

**Comment by francoismassart** [comment]
If you still get the runtime error try this patched version

`npm i eslint-plugin-tailwindcss@3.15.1-beta.0`

**Comment by andriibyk** [comment]
Hi @francoismassart 

The same type of issue is for `tailwindcss/no-unnecessary-arbitrary-value"` rule for patched version `"eslint-plugin-tailwindcss@3.15.1-beta.0"`

![image](https://github.com/francoismassart/eslint-plugin-tailwindcss/assets/8555142/1494eee8-a4d7-4a5b-81b9-d9db9ea1d555)

`index.tsx:41`
```diff
- <span className="!absolute !whitespace-nowrap !border-0 !p-0 ![clip:rect(0,0,0,0)]">Loading...</span>
+ <span className="!absolute !whitespace-nowrap !border-0 !p-0  [clip:rect(0,0,0,0)]">Loading...</span>
```

Once removed `!` for `![clip:rect(0,0,0,0)]` value - it does not cause an error

**Comment by francoismassart** [comment]
Thanks for the infos, I'll take a look at it

**Comment by francoismassart** [comment]
@coltonhurst please try with
`npm i eslint-plugin-tailwindcss@3.15.1-beta.1`

**Comment by francoismassart** [comment]
fixed in `eslint-plugin-tailwindcss@3.15.1`

**Comment by francoismassart** [comment]
@coltonhurst please re-open the issue if `eslint-plugin-tailwindcss@3.15.1` does not fix your issue

**Comment by coltonhurst** [comment]
@francoismassart apologies for the late reply. Thank you so much! This is fixed, I've tested `3.15.1` on [this](https://github.com/bitwarden/clients/pull/8388) PR. Thank you for the support!
